presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -1147,6 +1147,80 @@ private void testInsertPartitionedTableExistingPartition(Session session, HiveSt
     }
 
     @Test
+    public void testPartitionPerScanLimit()
+            throws Exception
+    {
+        TestingHiveStorageFormat storageFormat = new TestingHiveStorageFormat(
+                getSession(),
+                HiveStorageFormat.DWRF);
+        testPartitionPerScanLimit(storageFormat.getSession(), storageFormat.getFormat());
+    }
+
+    public void testPartitionPerScanLimit(Session session, HiveStorageFormat storageFormat)
+            throws Exception
+    {
+        String tableName = "test_partition_per_scan_limit";
+
+        @Language("SQL") String createTable = "" +
+                "CREATE TABLE " + tableName + " " +
+                "(" +
+                "  foo VARCHAR," +
+                "  part BIGINT" +
+                ") " +
+                "WITH (" +
+                "format = '" + storageFormat + "', " +
+                "partitioned_by = ARRAY[ 'part' ]" +
+                ") ";
+
+        assertUpdate(session, createTable);
+
+        TableMetadata tableMetadata = getTableMetadata(catalog, TPCH_SCHEMA, tableName);
+        assertEquals(tableMetadata.getMetadata().getProperties().get(STORAGE_FORMAT_PROPERTY), storageFormat);
+        assertEquals(tableMetadata.getMetadata().getProperties().get(PARTITIONED_BY_PROPERTY), ImmutableList.of("part"));
+
+        // insert 1200 partitions
+        for (int i = 0; i < 12; i++) {
+            assertUpdate(
+                    session,
+                    format(
+                            "INSERT INTO " + tableName + " " +
```

Comment:
Do this as `"INSERT INTO %s"` so that you're not mixing string concatenation with formatting